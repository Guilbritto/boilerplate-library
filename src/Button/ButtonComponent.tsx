import React from "react";
import { Container } from "./ButtonComponent.styles";
import { ButtonProps } from "./ButtonComponent.types";
import { ArrowRight } from "@svg/ArrowRight";
import { withTheme } from "styled-components";

const ButtonComponent = ({ icon, iconSide, label }: ButtonProps) => {
  const ButtonIcons = {
    arrowRight: ArrowRight,
  };

  return (
    <Container>
      {icon && iconSide === "left" && ButtonIcons[icon]}
      {label}
      {icon && iconSide === "right" && ButtonIcons[icon]}
    </Container>
  );
};

export default withTheme(ButtonComponent);
