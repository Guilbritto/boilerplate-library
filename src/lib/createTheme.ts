import { defaultTheme } from "../styles/defaultTheme";

export const createTheme = (theme: typeof defaultTheme) => {
  return {
    ...defaultTheme,
    colors: {
      ...defaultTheme.colors,
      ...theme.colors,
    },
    fontSizes: {
      ...defaultTheme.fontSizes,
      ...theme.fontSizes,
    },
  } as typeof defaultTheme;
}