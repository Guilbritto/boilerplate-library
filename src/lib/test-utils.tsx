import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from '../styles/defaultTheme'

const Providers: FC = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
