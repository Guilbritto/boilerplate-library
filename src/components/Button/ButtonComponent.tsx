import React from 'react'
import { Container, DotContaier } from './ButtonComponent.styles'
import { ButtonProps } from './ButtonComponent.types'
import { ArrowRight } from '@svg/ArrowRight'
import { useTheme } from '../../hooks/useTheme'
import {BiRightArrowAlt} from 'react-icons/bi'

const ButtonComponent = ({
    icon,
    iconSide='right',
    label,
    loading,
    marginBottom,
    marginLeft,
    marginRight,
    marginTop,
    width,
    buttonSize="large",
    buttonType="primary",
    disabled,
    ...rest
}: ButtonProps) => {
    const theme = useTheme()
    const ButtonIcons = {
        arrowRight: <BiRightArrowAlt />
    }

    return (
        <Container
            theme={theme}
            isDisabled={loading ? true : disabled}
            buttonSize={buttonSize}
            buttonType={disabled ? 'disabled' : buttonType}
            style={{
                marginBottom,
                marginLeft,
                marginRight,
                marginTop,
                width
            }}
            disabled={loading ? true : disabled}
            {...rest}
        >
            {loading ? (
                <>
                    <DotContaier>
                        <span className="dot" key={`dot_1`} />
                        <span className="dot" key={`dot_2`} />
                        <span className="dot" key={`dot_3`} />
                    </DotContaier>
                </>
            ) : (
                <>
                    {icon && iconSide === 'left' && ButtonIcons[icon]}
                    {label}
                    {icon && iconSide === 'right' && ButtonIcons[icon]}
                </>
            )}
        </Container>
    )
}

export default ButtonComponent
