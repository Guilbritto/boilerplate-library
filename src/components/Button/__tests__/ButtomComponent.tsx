import ButtonComponent from '../ButtonComponent'
import { render } from '../../../lib/test-utils'

describe('Button', () => {
    it('Should be able pass to snapshot test', () => {
        const { container } = render(
            <ButtonComponent label="Teste" iconSide="left" icon="arrowRight" />
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
