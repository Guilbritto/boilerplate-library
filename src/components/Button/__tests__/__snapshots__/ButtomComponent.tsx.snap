// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should be able pass to snapshot test 1`] = `
<button
  class="sc-bdfBQB knuzpq"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
    />
  </svg>
  Teste
</button>
`;

exports[`Button Should render a danger version 1`] = `
<button
  class="sc-bdfBQB eyCNfi"
>
  Teste
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
    />
  </svg>
</button>
`;

exports[`Button Should render a disabled button 1`] = `
<button
  class="sc-bdfBQB grbVLB"
  disabled=""
>
  Teste
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
    />
  </svg>
</button>
`;

exports[`Button Should render a secondary version 1`] = `
<button
  class="sc-bdfBQB dpPrOe"
>
  Teste
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
    />
  </svg>
</button>
`;

exports[`Button Should render a small size 1`] = `
<button
  class="sc-bdfBQB buALzq"
>
  Teste
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
    />
  </svg>
</button>
`;

exports[`Button Should render the button with the on the right side 1`] = `
<button
  class="sc-bdfBQB knuzpq"
>
  Teste
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
    />
  </svg>
</button>
`;
